File: backend/src/EquipmentManagement.API/Domain/Entities/Equipment.cs
using EquipmentManagement.API.Domain.Enums;

namespace EquipmentManagement.API.Domain.Entities;

public class Equipment
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string SerialNumber { get; set; } = string.Empty;
    public DateTime AcquisitionDate { get; set; }
    public EquipmentStatus Status { get; set; } = EquipmentStatus.Active;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/equipment.ts
export type EquipmentStatus = 'Active' | 'Inactive' | 'UnderMaintenance' | 'Decommissioned';

export interface Equipment {
  id: string;
  name: string;
  type: string;
  serialNumber: string;
  acquisitionDate: string;
  status: EquipmentStatus;
  createdAt: string;
  updatedAt: string;
}

export interface EquipmentCreateRequest {
  name: string;
  type: string;
  serialNumber: string;
  acquisitionDate: string;
  status: EquipmentStatus;
}

export interface EquipmentUpdateRequest {
  name: string;
  type: string;
  serialNumber: string;
  acquisitionDate: string;
  status: EquipmentStatus;
}

export interface EquipmentQuery {
  page?: number;
  pageSize?: number;
  search?: string;
  type?: string;
  status?: string;
  sortBy?: string;
  sortDirection?: 'asc' | 'desc';
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
}

export const EQUIPMENT_STATUS_LABELS: Record<EquipmentStatus, string> = {
  Active: 'Ativo',
  Inactive: 'Inativo',
  UnderMaintenance: 'Em Manutenção',
  Decommissioned: 'Desativado',
};

export const EQUIPMENT_STATUS_OPTIONS: { value: EquipmentStatus; label: string }[] = [
  { value: 'Active', label: 'Ativo' },
  { value: 'Inactive', label: 'Inativo' },
  { value: 'UnderMaintenance', label: 'Em Manutenção' },
  { value: 'Decommissioned', label: 'Desativado' },
];
